### DISCUSSION TITLE
Project JavaScript Version

### DISCUSSION THREAD

**Discussion by outplex** [amazon-connect/voicemail-express-amazon-connect#24] [General]
Hi guys,

I'm using this repo to create a solution for one of my clients, you've done an amazing job, I appreciate your effort. However, I was using the previous version of this solution written in JavaScript and I just noticed that the entire project was re-written in Python, which is great. But my question is if we could leave the older version available somewhere (maybe in another repository or branch), that way we can continue to build/improve our enxisting code and we're not forced to entirely rewrite our solutions to python.

I'm relatively new to this, my apologies in advanced if this is something that is not necessary or if there's a viable solution already in place.

**Comment by dougjaso** [comment]
A couple of notes: 

The previous version had 1 Javascript function, which was the KVS to S3 version. It was written using a realtime API to access data from Kinesis and had a problem that could cause the lambda function to timeout if there was a significant amount of traffic on the Conenct instance using KVS. If yoiu can upgrade, we'd strongly recommend it.

If not, you can always get to previous versions by going back through the history. Here is the last version of the [Javascript function](https://github.com/amazon-connect/voicemail-express-amazon-connect/blob/ef6f44679529496ac2155e4dbb0f539065b8511a/Code/Core/vmx3_kvs_to_s3.js).

Additionally, if you need to load a previous version, you can just modify the launch URL to get the version you want. For example:


https://console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/new?stackName=VMX3&templateURL=https://vmx-source-us-west-2.s3.us-west-2.amazonaws.com/vmx3/2024.08.01/cloudformation/vmx3.yaml

In there, just modify the region (in two places) to match your Connect region, and the version to match the version (2024.08.01 in this example) that you want, and it will run. We do not remove the old versions from S3.
